08.ls_object/contents_producer.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+module ContentsProducer
+  private
+
+  def fetch_contents(input_target, options)
+    target_contents = []
+    return [input_target] if FileTest.file?(input_target)
+
+    return unless Dir.exist?(input_target)
```

Comment:
`nil` を返したいなら、それが分かるように明示的に書いたほうが良さそうです。

```suggestion
    return nil unless Dir.exist?(input_target)
```